qiskit/transpiler/__init__.py

```diff
@@ -34,84 +34,164 @@
-computational effort does not always result in an improvement in the quality of the output
-circuit.
+Qiskit uses the graph-based :class:`.DAGCircuit` intermediate representation (IR) of a circuit
+throughout the transpiler stack, rather than the tree-based :class:`.QuantumCircuit`.  A transpiler
+pipeline is a :class:`.PassManager` object, whose :meth:`.PassManager.run` method takes in a
+:class:`.QuantumCircuit`, converts it to a :class:`.DAGCircuit`, then subjects the IR to a sequence
+of *passes*, finally returning a :class:`.QuantumCircuit` back.  A pass is either an
+:class:`.AnalysisPass`, whose purpose is to calculate and store properties about the circuit in the
+stateful :class:`.PropertySet`, or a :class:`.TransformationPass`, whose purpose is to modify the IR
+to achieve a particular singular goal.  We typically think of a pipeline as being split into
+"stages", where each stage is responsible for one high-level transformation.
+
+Qiskit exposes a default transpilation pipeline builder via the function
+:func:`.generate_preset_pass_manager`.  This will return a properly configured pipeline for complete
+transpilation, at a chosen ``optimization_level`` (between 0 and 3, inclusive).  Unless you are
+looking for something highly specialized, this is almost certainly the entry point you want.  A
+sample transpilation looks like::
+
+    from qiskit.circuit import QuantumCircuit
+    from qiskit.transpiler import generate_preset_pass_manager
+    from qiskit_ibm_runtime import QiskitRuntimeService
+
+    # Any abstract circuit you want:
+    abstract = QuantumCircuit(2)
+    abstract.h(0)
+    abstract.cx(0, 1)
+
+    # Any method you like to retrieve the backend you want to run on:
+    backend = QiskitRuntimeService().backend("some-backend")
+
+    # Create the pass manager for the transpilation ...
+    pm = generate_preset_pass_manager(backend=backend)
+    # ... and use it (as many times as you like).
+    physical = pm.run(abstract)
+
+For most use cases, this is all you will need.
+All of Qiskit's transpiler infrastructure is highly extensible and configurable, however.
+The rest of this page details how to harness the low-level capabilities of the transpiler stack.
+
+.. _transpiler-preset:
+
+Preset Pass Managers
+====================
 
-If you'd like to work directly with a
-preset pass manager you can use the :func:`~.generate_preset_pass_manager`
-function to easily generate one. For example:
+The function :func:`.generate_preset_pass_manager` creates the "preset pass managers".
+These are all instances of :class:`.PassManager`, so are used by pass a :class:`.QuantumCircuit` to
+the :meth:`.PassManager.run` method.  More specifically, the preset pass managers are instances of
+:class:`.StagedPassManager`, which allows greater configuration of the individual stages of a
+tranpsilation.
+
+A preset pass manager has up to six named stages.  These are summarized, in order of execution,
```

Comment:
I think we should also briefly mention the ability to add `pre_` and `post_` stages, maybe as a note at the end of this section? The `pre_layout` stage is used later in an example but I didn't find it being introduced anywhere in the text.

Reply:
Shall I put that documentation into the bit about `StagedPassManager` further down the page?

Imo the preset pass managers shouldn't use the `pre_` and `post_` sections themselves at all, since that defeats their purpose for configuration; if we use them, we've de facto made that its own stage.  But I think we need to re-jig the construction a little bit to make that true; at the moment, we _do_ use them.

Reply:
Done in adacfe9ab7.